### PR TITLE
Remove stale templates and styles

### PR DESCRIPTION
**Pull request by mtruj013**
## Done

- Removed stale templates and corresponding styles
- Drive-by's:
  - Updated partners site map
  - Fixed indentation 
  - Added redirects

## Notes (please read before reviewing)

There are four stale surus that are still in use and as such were not removed in this PR:

- `p-strip--suru`, used in https://canonical-com-978.demos.haus/blog/archives 
- `p-strip--suru-topped` used in /careers/thank-you:
Screenshot from 2023-07-19 18-31-35
- `p-strip--sutu-top` used in https://canonical-com-978.demos.haus/partners/request-login
- `p-strip--suru-background` used in https://canonical-com-978.demos.haus/partners/isv-partners

I believe these pages are still needed so will update them according to updated designs in a separate PR.

There are also two stale base templates that are still needed for now:
-  `/careers/base-template` 
- `/careers/base-tabs`

To give some context,  these used to be the base templates for the old department pages/all roles page which included almost all of careers content as tabbed content. The department pages should not extend either if these anymore, but in order to update this the back end views which render these templates need to be refactored. I'll do this in a separate PR as well. 

## QA 

- Check out this feature branch
- Run the site using the command `./run serve`
- Visit the following list of removed templates and see that you are properly redirected: 
  - https://canonical-com-978.demos.haus/careers/ethics 
  - https://canonical-com-978.demos.haus/careers/lifestyle
  - https://canonical-com-978.demos.haus/partners/apps-and-snaps
  - https://canonical-com-978.demos.haus/partners/devices-and-iot
  - https://canonical-com-978.demos.haus/partners/isv-partners
  - https://canonical-com-978.demos.haus/partners/kubernetes
- Check that removing the following partials does not affect existing pages. There should be no visual or functional changes when compared against the live site:
  - `/careers/_vacancies.modal.html` check against careers pages and career explorer on live site. 

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-5153

### CODE REVIEW

**Comment by juanruitina** [comment]
The IoT page redirects to "/partners/partners/iot-device" instead of "/partners/iot-device", all others seem alright. I'll double check if we are good deleting these partner pages, in any case.

**Comment by mtruj013** [comment]
Thanks @juanruitina, the redirect should be fixed now

**Comment by juanruitina** [comment]
Please, don't remove nor redirect the ISV page yet: we will keep it, although we need to rebrand it. I suggest that we take care of this on a separate PR to not block all the rest.

I have noticed the page `title` for /partners is the same as for the home, can you change it for "Canonical partner programmes"? Other than this, changes LGTM, feel free to UX+1 this once you address my comments (ignore the title thing if you would rather handle it in a separate PR).

File: templates/partners/request-login.html [hunk 1]
```diff
@@ -154,4 +154,4 @@ <h1>Find a partner</h1>
       }
     </style>
 
-    {% endblock %}
+    {% endblock %}
```
Comment (carkod) [inline templates/partners/request-login.html:157]: Is this change needed?

**Review by carkod — APPROVED** [review]
LGTM

File: templates/partners/request-login.html (hunk 1 above)
Comment (mtruj013) [reply templates/partners/request-login.html:157]: It wasn't intentional, I'd originally removed this page but I had to add it back. Extra line got added when I copy/pasted it back in
